tests/test_translator.py

```
@@ -0,0 +1,1198 @@
+        # So regex is tried first, which matches 'a' -> 'a'
+        self.assertEqual(result.first('a'), 'a')  # regex matches first
+        self.assertEqual(result.first('b'), 'b')  # regex matches
+
+
+class TestSelfTranslatorCoverage(unittest.TestCase):
+
+    def test_self_first_with_single_string(self) -> None:
+        """Test first() with a single string."""
+        trans = SelfTranslator()
+        result = trans.first('test')
+        self.assertEqual(result, 't')  # strings[0] of 'test' is 't'
```

Ruff (0.14.5)

Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)